website/content/docs/ecs/architecture.mdx

```diff
@@ -39,6 +39,142 @@ This diagram shows the timeline of a task starting up and all its containers:
 - **T1:** The `sidecar-proxy` container starts. It runs Envoy by executing `envoy -c <path-to-bootstrap-json>`.
 - **T2:** The `sidecar-proxy` container is marked as healthy by ECS. It uses a health check that detects if its public listener port is open. At this time, your application containers are started since all Consul machinery is ready to service requests. The only running containers are `consul-client`, `sidecar-proxy`, and your application container(s).
 
+### Task Shutdown
+
+Graceful shutdown is supported when deploying Consul on ECS, which means the following:
+
+* No incoming traffic from the mesh is directed to this task during shutdown.
+* Outgoing traffic to the mesh is possible during shutdown.
+
+This diagram shows an example timeline of a task shutting down:
+
+<img alt="Task Shutdown Timeline" src="/img/ecs-task-shutdown.svg" style={{display: "block", maxWidth: "400px"}} />
+
+- **T0**: ECS sends a TERM signal to all containers.
+  - `consul-client` begins to gracefully leave the Consul cluster, since it is configured with `leave_on_terminate = true`
+  - `health-sync` stops syncing health status from ECS into Consul checks.
+  - `sidecar-proxy` ignores the TERM signal and continues running until it notices that `user-app` container has exited. This allows the application container to continue to make outgoing requests through the proxy to the mesh. This possible due to an entrypoint override for the container, `consul-ecs envoy-entrypoint`.
+  - `user-app` exits since, in this example, it does not intercept the TERM signal
+- **T1**:
+  - `health-sync` updates its Consul checks to critical status, and then exits. This ensures this service instance is marked unhealthy.
+  - The `sidecar-proxy` container checks the ECS task metadata. It notices the `user-app` container has stopped, and exits.
+- **T2**:
+  - `consul-client` finishes leaving the Consul cluster and exits
+  - Updates about this task have reached the rest of the Consul cluster, which means downstream proxies are updated to stop sending traffic to this task.
+- **T3**: All containers have exited
+  - `consul-client` finishes gracefully leaving the Consul datacenter and exits.
+  - ECS notices all containers have exited, and will soon put change the Task status to `STOPPED`
+- **T4**: (Not applicable to this example, but if any conatiners are still running at this point, ECS forcefully stops them by sending a KILL signal)
+
+#### Task Shutdown: Completely Avoiding Application Errors
+
+Because Consul service mesh is a distributed, eventually consistent system that is subject to network latency, it is hard to achieve a perfect graceful shutdown.
+
+In particular, you may have noticed the following issue in example above, where it is possible that an application that has exited still receives incoming traffic:
```

Comment:
```suggestion
In some cases, an exited application can receive incoming traffic. The following example of this behavior draws on the timeline described above:
```
It wasn't really clear that the process is actually an example. Also, how could the user experience the behavior if we didn't provide any instructions to specifics to walk through the example? It may be less confusing to assume that the user hasn't walked through a tutorial or procedure.